src/main/src/main/java/org/geoserver/catalog/LayerGroupHelper.java

```diff
@@ -531,6 +543,29 @@ private static void expandStyleGroup(
         }
     }
 
+    /**
+     * Ensures that WMSLayer Info inside layergroup associates with correct remote style
+     *
+     * @return returns a compatible StyleInfo representing a remote style if required
+     */
+    private static StyleInfo processWMSLayer(WMSLayerInfo wmsLayerInfo, StyleInfo styleInfo) {
+
+        if (styleInfo == null) styleInfo = wmsLayerInfo.getDefaultStyle();
```

Comment:
I find this method hard to read:
- Why is named process layer? it doesn't seem to process anything specific, it just retrieves the correct style for a remote layer no? so why not name it something like ``getRemoteWmsLayerStyle``.
- What's the provided ``styleInfo`` object?

Reply:
understood will change name.

The provided style is whatever style was configured against the layer during layergroup configuration. In case of cascaded wms layer, its the default raster